Google.PowerShell/Compute/GceMetadataCmdlet.cs

```diff
@@ -0,0 +1,173 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
+// Licensed under the Apache License Version 2.0.
+
+using Google.PowerShell.Common;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Management.Automation;
+using System.Net;
+
+namespace Google.PowerShell.ComputeEngine
+{
+    /// <summary>
+    /// <para type="synopsis">
+    /// Gets the current metadata from the metadata server.
+    /// </para>
+    /// <para type="description">
+    /// Gets the current metadata from the metadata server. Get-GceMetadata can only be called from a Google
+    /// Compute Engine VM instance. Calls from any other machine will fail.
+    /// </para>
+    /// <para type="link" uri="(https://cloud.google.com/compute/docs/storing-retrieving-metadata)">
+    /// [Metadata Documentation]
+    /// </para>
+    /// <example>
+    /// <code>PS C:\> $allMetadata = Get-GceMetadata -Recursive | ConvertFrom-Json</code>
+    /// <para>Gets all of the metadata and converts it from the given JSON to a PSCustomObject.</para>
+    /// </example>
+    /// <example>
+    /// <code>PS C:\> $hostName = Get-GceMetadata -Path "instance/hostname" </code>
+    /// <para>Gets the hostname of the instance.</para>
+    /// </example>
+    /// <example>
+    /// <code>PS C:\> $customProjectMetadata = Get-GceMetadata -Path "project/attributes/customKey" </code>
+    /// <para>Gets the value of the custom metadata with key "customKey" placed in the project .</para>
+    /// </example>
+    /// <example>
+    /// <code>PS C:\> $metadata, $etag = Get-GceMetadata -AppendETag -Recursive</code>
+    /// <para>Gets the entire metadata tree, and the ETag of the version retrieved.</para>
+    /// </example>
+    /// <example>
+    /// <code>
+    ///     PS C:\> $newTags, $newEtag = Get-GceMetadata -Path "instance/tags" -AppendETag -WaitUpdate `
+    ///                                     -LastETag $oldETag
+    /// </code>
+    /// <para>Waits for the metadata "instance/tags" to be updated by the server.</para>
+    /// </example>
+    /// </summary>
+    [Cmdlet(VerbsCommon.Get, "GceMetadata")]
+    [OutputType(typeof(string))]
+    public class GetGceMetadataCmdlet : GCloudCmdlet
+    {
+        /// <summary>
+        /// <para type="description">
+        /// The path to the specific metadata you wish to get e.g. "instance/tags", "instance/attributes",
+        /// "project/attributes/sshKeys".
+        /// </para>
+        /// </summary>
+        [Parameter]
+        public string Path { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// If set, will get the metadata subtree as a JSON string. If -Path is not set, will get the entire
+        /// metadata tree as a JSON string.
+        /// </para>
+        /// </summary>
+        [Parameter]
+        public SwitchParameter Recursive { get; set; }
```

Comment:
I'm not saying I like this, but `-Recurse` is what `Get-ChildItem` uses.
https://technet.microsoft.com/en-us/library/hh849800.aspx